### PR TITLE
add the aad migration tests for the current aad migration script

### PR DESCRIPTION
**Pull request by hckenmiller**

### CODE REVIEW

File: Fabric.Identity.API/scripts/tests/Install-Identity-Utilities.tests.ps1 [hunk 1]
```diff
@@ -874,3 +874,194 @@ Describe 'Confirm-DiscoveryConfig Unit Tests'{
         }
     }
 }
+
+Describe 'Migrate-AADSettings' -Tag 'Unit'{
+    Context 'Migrating AAD Settings using Unit Tests'{
+        InModuleScope Install-Identity-Utilities{
+            It 'Should successfuly run the migration'{
```
Comment (joshnielsen10) [inline Fabric.Identity.API/scripts/tests/Install-Identity-Utilities.tests.ps1]: *successfully

File: Fabric.Identity.API/scripts/tests/Install-Identity-Utilities.tests.ps1 [hunk 2]
```diff
@@ -874,3 +874,194 @@ Describe 'Confirm-DiscoveryConfig Unit Tests'{
...
+                 # Add them to the powershell TestDrive which cleans up after each test leaving the tests folder clean
+                 $testInstallFile = 'testInstall.config'
+                 $testAzureFile = 'testAzure.config'
+                 $installConfigPath = "$($TestDrive)\$($testInstallFile)"
+                 $azureConfigPath = "$($TestDrive)\$($testAzureFile)"
+                 $nodesToSearch = @("tenants","replyUrls","claimsIssuerTenant","allowedTenants","registeredApplications", "azureSecretName")
+
+                 $doesInstallFileExist = Test-Path $installConfigPath
+                 $doesAzureFileExist = Test-Path $azureConfigPath
+                 if (!$doesInstallFileExist)
```
Comment (ctediz) [inline Fabric.Identity.API/scripts/tests/Install-Identity-Utilities.tests.ps1]: I think it's possible to pull out this logic into a BeforeEach section
https://github.com/pester/Pester/wiki/BeforeEach-and-AfterEach

File: Fabric.Identity.API/scripts/tests/Install-Identity-Utilities.tests.ps1 [hunk 3]
```diff
@@ -874,3 +874,194 @@ Describe 'Confirm-DiscoveryConfig Unit Tests'{
...
+Describe 'Migrate-AADSettings' -Tag 'Integration'{
+    Context 'Migrating AAD Settings using Integration Tests'{
+        InModuleScope Install-Identity-Utilities{
+            It 'Should successfully run the migration'{
+                 # Arrange 
+                 # Get the contents of the files for testing from the tests folder
+                 # Add them to the powershell TestDrive which cleans up after each test leaving the tests folder clean
+                 $testInstallFile = 'testInstall.config'
+                 $testAzureFile = 'testAzure.config'
+                 $installConfigPath = "$($TestDrive)\$($testInstallFile)"
```
Comment (ctediz) [inline Fabric.Identity.API/scripts/tests/Install-Identity-Utilities.tests.ps1]: What is TestDrive? is that something built into pester?

File: Fabric.Identity.API/scripts/tests/testInstall.config [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+        <variable name="4d07d6d8-58e4-45a4-8ce9-5d2cfc00c65f" alias="MikeTenant" />
+      </claimsIssuerTenant>
+      <!-- The Tenants that are allowed through to the claims issuer  -->
+      <allowedTenants>
+        <variable name="0ced5a42-fa6e-4d17-bb75-da8c4dbec43c" alias="RyanTenant" />
+      </allowedTenants>
+      <!-- The name of the secret to be created for the azure application registration -->
+      <variable name="azureSecretName" value="AADOnceAgain" />
+      <registeredApplications>
+        <variable appName="Identity Provider Search Service" tenantId="4d07d6d8-58e4-45a4-8ce9-5d2cfc00c65f" tenantAlias="MikeTenant" clientid="2057a221-b20a-4491-9170-9638883e6af0" secret="tTSDjz3ksRYXtDRilAWuzdjIwQ8Sd1kjI2H2H8NtQwU=" />
```
Comment (ctediz) [inline Fabric.Identity.API/scripts/tests/testInstall.config]: Swap these with dummy values instead of actual values

Comment (hckenmiller) [reply Fabric.Identity.API/scripts/tests/testInstall.config]: yup

File: Fabric.Identity.API/scripts/tests/Install-Identity-Utilities.tests.ps1 (hunk 3 above)
Comment (hckenmiller) [reply Fabric.Identity.API/scripts/tests/Install-Identity-Utilities.tests.ps1]: built into powershell. https://github.com/pester/Pester/wiki/TestDrive

File: Fabric.Identity.API/scripts/tests/Install-Identity-Utilities.tests.ps1 (hunk 2 above)
Comment (hckenmiller) [reply Fabric.Identity.API/scripts/tests/Install-Identity-Utilities.tests.ps1]: will look into.

File: Fabric.Identity.API/scripts/tests/Install-Identity-Utilities.tests.ps1 [hunk 5]
```diff
@@ -874,3 +874,194 @@ Describe 'Confirm-DiscoveryConfig Unit Tests'{
...
+                Assert-MockCalled -ModuleName Install-Identity-Utilities -CommandName Search-XMLChildNode -Scope It -Times 1 -Exactly
+                Assert-MockCalled -ModuleName Install-Identity-Utilities -CommandName Remove-XMLChildNodes -Scope It -Times 2 -Exactly
+                Assert-MockCalled -ModuleName Install-Identity-Utilities -CommandName Add-XMLChildNodes -Scope It -Times 1 -Exactly
+                Assert-MockCalled -ModuleName Install-Identity-Utilities -CommandName Compare-Object -Scope It -Times 1 -Exactly
+            }
+        }
+    }
+}
+Describe 'Migrate-AADSettings' -Tag 'Integration'{
+    Context 'Migrating AAD Settings using Integration Tests'{
```
Comment (miketabet) [inline Fabric.Identity.API/scripts/tests/Install-Identity-Utilities.tests.ps1]: Consider a few more test cases for edge cases - no config file found, no permissions to the config file, and malformed XML file.  Adding these can also help drive improved error messaged.

Comment (hckenmiller) [reply Fabric.Identity.API/scripts/tests/Install-Identity-Utilities.tests.ps1]: I considered the no config file, but I believe I would have to do it outside of pester, but didn't have time yet to do that. Pester tests require functions vs just running a powershell script, because functions (isolated code) are easy to test. I think the other two, no permissions and malformed XML, are also good ideas. I will see what I can do.
